app/jobs/in_person/send_proofing_notification_job.rb

```diff
@@ -6,75 +6,66 @@ class SendProofingNotificationJob < ApplicationJob
-              enrollment_id: enrollment.id,
-            )
-          return
-        end
-        analytics(user: enrollment.user).
-          idv_in_person_usps_proofing_results_notification_job_started(
-            enrollment_code: enrollment.enrollment_code,
-            enrollment_id: enrollment.id,
+
+      enrollment = InPersonEnrollment.find_by(
+        { id: enrollment_id },
+        include: [:notification_phone_configuration, :user],
+      )
+
+      if enrollment.nil? || !enrollment.eligible_for_notification?
+        analytics(user: enrollment&.user || AnonymousUser.new).
+          idv_in_person_usps_proofing_results_notification_job_skipped(
+            enrollment_code: enrollment&.enrollment_code,
+            enrollment_id: enrollment&.id,
           )
-        if enrollment.expired?
-          # no sending message for expired status
-          enrollment.notification_phone_configuration&.destroy
-          return
-        end
+        return
+      end
 
-        # only send sms when success or failed
-        # send notification and log result
-        phone = enrollment.notification_phone_configuration.formatted_phone
-        message = notification_message(enrollment: enrollment)
-        response = Telephony.send_notification(
-          to: phone, message: message,
-          country_code: Phonelib.parse(phone).country
+      analytics(user: enrollment.user).
+        idv_in_person_usps_proofing_results_notification_job_started(
+          enrollment_code: enrollment.enrollment_code,
+          enrollment_id: enrollment.id,
         )
-        handle_telephony_result(enrollment: enrollment, phone: phone, telephony_result: response)
-        # if notification sent successful
-        enrollment.update(notification_sent_at: Time.zone.now) if response.success?
-      ensure
-        Rails.logger.error("Unknown enrollment with id #{enrollment_id}") unless enrollment.present?
-        analytics(user: enrollment.present? ? enrollment.user : AnonymousUser.new).
-          idv_in_person_usps_proofing_results_notification_job_completed(
-            enrollment_code: enrollment&.enrollment_code, enrollment_id: enrollment_id,
-          )
+      if enrollment.expired?
+        # no sending message for expired status
+        enrollment.notification_phone_configuration&.destroy
+        log_job_completed(enrollment: enrollment)
+        return
       end
+
+      # send notification and log result when success or failed
+      phone = enrollment.notification_phone_configuration.formatted_phone
+      message = notification_message(enrollment: enrollment)
+      response = Telephony.send_notification(
+        to: phone, message: message,
+        country_code: Phonelib.parse(phone).country
+      )
+      handle_telephony_response(enrollment: enrollment, phone: phone, telephony_response: response)
+
+      enrollment.update(notification_sent_at: Time.zone.now) if response.success?
+
+      log_job_completed(enrollment: enrollment)
```

Comment:
The removal of the `ensure` section here seems like it could hinder debugging - i.e. we won't have the ID of enrollments for which this job fails upfront, so more work may be required in terms of log correlation.

Reply:
we will still have either 'skipped' or 'started' events for every enrollment passed to the job.

Reply:
Not in the case of a database exception, though.

Reply:
`find_by` is used instead of `find` because it won't raise an exception if the record isn't found.

```ruby
InPersonEnrollment.includes(:notification_phone_configuration, :user).find(999)
=> ActiveRecord::RecordNotFound: Couldn't find InPersonEnrollment with 'id'=999

InPersonEnrollment.find_by({ id: 999 }, include: [:notification_phone_configuration, :user])
=> nil
```

Reply:
I am aware of that case - I'm talking here about the more general case where the database connection is interrupted.

Reply:
addressed in 9ea8ca7ae0690cde7b86bd3a9a0282c1ea93adc9